### ISSUE TITLE
Drop minio/sha256-simd in favor of crypto/sha256 for Go 1.21+

### ISSUE THREAD

**Issue by liamsi** [celestiaorg/celestia-app#1826]
see: https://github.com/multiformats/go-multihash/pull/173

please also open issues in other repos (core, maybe nmt etc)

**Comment by rootulp** [comment]
```
$ go mod why github.com/minio/sha256-simd
# github.com/minio/sha256-simd
github.com/celestiaorg/celestia-app/pkg/appconsts
github.com/celestiaorg/rsmt2d
github.com/minio/sha256-simd
```

so I created https://github.com/celestiaorg/rsmt2d/issues/161

**Comment by rootulp** [comment]
> please also open issues in other repos (core, maybe nmt etc)

Just checked and
1. celestia-core doesn't depend on minio/sha256-simd. See https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/go.mod
2. nmt doesn't depend on minio/sha256-simd. See https://github.com/celestiaorg/nmt/blob/master/go.mod

**Comment by tac0turtle** [comment]
this is gone from v4, only in test/interchain/go.mod does it exist. should it stay open for solving that part?

**Comment by rootulp** [comment]
Agreed I don't see `minio/sha256-simd` in the v4 go.mod. There are other minio deps like:

```
	github.com/minio/highwayhash v1.0.3 // indirect
	github.com/minio/md5-simd v1.1.2 // indirect
	github.com/minio/minio-go/v7 v7.0.74 // indirect
```

FWIW I don't think this was resolved by the v4 mega PR because the minio deps weren't changed in https://github.com/celestiaorg/celestia-app/pull/4400/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L173-L175 but seems safe to close.
